tests/ops/service/privacy_request/request_runner_service_test.py

```diff
@@ -1773,3 +1773,91 @@ def test_email_connector_no_updates_needed(
         assert (
             mailgun_send.called is False
         ), "Email not sent because no updates are needed. Data category doesn't apply to any of the collections."
+
+
+class TestPrivacyRequestsManualWebhooks:
+    @mock.patch("fidesops.ops.service.privacy_request.request_runner_service.upload")
+    def test_privacy_request_needs_manual_input(
+        self,
+        mock_upload,
+        integration_manual_webhook_config,
+        access_manual_webhook,
+        policy,
+        run_privacy_request_task,
+        db,
+    ):
+        customer_email = "customer-1@example.com"
+        data = {
+            "requested_at": "2021-08-30T16:09:37.359Z",
+            "policy_key": policy.key,
+            "identity": {"email": customer_email},
+        }
+
+        pr = get_privacy_request_results(
+            db,
+            policy,
+            run_privacy_request_task,
+            data,
+        )
+        db.refresh(pr)
+        assert pr.status == PrivacyRequestStatus.requires_input
+        assert not mock_upload.called
+
+    @mock.patch("fidesops.ops.service.privacy_request.request_runner_service.upload")
+    @mock.patch(
+        "fidesops.ops.service.privacy_request.request_runner_service.run_erasure"
+    )
+    def test_manual_input_not_required_for_erasure_only_policies(
+        self,
+        mock_erasure,
+        mock_upload,
+        integration_manual_webhook_config,
+        access_manual_webhook,
+        erasure_policy,
+        run_privacy_request_task,
+        db,
+    ):
+        """Manual inputs are not tied to policies, but shouldn't hold up request if only erasures are requested"""
+        customer_email = "customer-1@example.com"
+        data = {
+            "requested_at": "2021-08-30T16:09:37.359Z",
+            "policy_key": erasure_policy.key,
+            "identity": {"email": customer_email},
+        }
+
+        pr = get_privacy_request_results(
+            db,
+            erasure_policy,
+            run_privacy_request_task,
+            data,
+        )
+        db.refresh(pr)
+        assert (
+            pr.status == PrivacyRequestStatus.complete
+        )  # Privacy request not put in "requires_input" state
+        assert not mock_upload.called  # erasure only request, no data uploaded
+        assert mock_erasure.called
+
+    @mock.patch("fidesops.ops.service.privacy_request.request_runner_service.upload")
+    def test_pass_on_manually_added_input(
```

Comment:
can we additionally test the case where only some of the data needed is input?

Reply:
yes I can better verify the behavior in these end-to-end tests, good call